### PR TITLE
Added serializing spell-specific data

### PR DESCRIPTION
**Pull request by Azami7**

### CODE REVIEW

File: Ollivanders/src/net/pottercraft/Ollivanders2/Ollivanders2Common.java [hunk 1]
```diff
@@ -1,20 +1,22 @@
 package net.pottercraft.Ollivanders2;
 
 import net.pottercraft.Ollivanders2.Spell.Spells;
-import org.bukkit.Location;
-import org.bukkit.Material;
+
+import org.bukkit.*;
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/Ollivanders2Common.java]: Do you want .* ?

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/COLLOPORTUS.java [hunk 2]
```diff
@@ -17,9 +20,38 @@ public COLLOPORTUS (Player player, Location location, StationarySpells name, Int
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/COLLOPORTUS.java]: why do the serialize and deserialize do nothing?

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/HORCRUX.java [hunk 3]
```diff
@@ -48,4 +57,25 @@ public void checkEffect (Ollivanders2 p)
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/HORCRUX.java]: same question here

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MOLLIARE.java [hunk 4]
```diff
@@ -17,8 +20,37 @@ public MOLLIARE (Player player, Location location, StationarySpells name, Intege
...
+
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MOLLIARE.java]: and here

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MUFFLIATO.java [hunk 5]
```diff
@@ -18,8 +20,37 @@ public MUFFLIATO (Player player, Location location, StationarySpells name, Integ
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MUFFLIATO.java]: yup

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_APPAREBIT.java [hunk 6]
```diff
@@ -17,8 +20,37 @@ public NULLUM_APPAREBIT (Player player, Location location, StationarySpells name
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_APPAREBIT.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_EVANESCUNT.java [hunk 7]
```diff
@@ -16,8 +19,37 @@ public NULLUM_EVANESCUNT (Player player, Location location, StationarySpells nam
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_EVANESCUNT.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/PROTEGO_HORRIBILIS.java [hunk 8]
```diff
@@ -40,4 +50,25 @@ public void checkEffect (Ollivanders2 p)
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/PROTEGO_HORRIBILIS.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/REPELLO_MUGGLETON.java [hunk 9]
```diff
@@ -68,4 +79,25 @@ else if (isInside(player.getLocation()))
...
+   /**
+    * Serialize all data specific to this spell so it can be saved.
+    *
+    * @param p unused for this spell
+    * @return a map of the serialized data
+    */
+   @Override
+   public Map<String, String> serializeSpellData (Ollivanders2 p)
+   {
+      return new HashMap<>();
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/REPELLO_MUGGLETON.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/COLLOPORTUS.java (hunk 2 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/COLLOPORTUS.java]: I have not written the deserialize yet. That is the next task. Step 1 is make sure we're writing the files correctly and nothing blows up.

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/HORCRUX.java (hunk 3 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/HORCRUX.java]: same answer

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MOLLIARE.java (hunk 4 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MOLLIARE.java]: same

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MUFFLIATO.java (hunk 5 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/MUFFLIATO.java]: same

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_APPAREBIT.java (hunk 6 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_APPAREBIT.java]: same

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_EVANESCUNT.java (hunk 7 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/NULLUM_EVANESCUNT.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/PROTEGO_HORRIBILIS.java (hunk 8 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/PROTEGO_HORRIBILIS.java]: ...

File: Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/REPELLO_MUGGLETON.java (hunk 9 above)
Comment (Azami7) [reply Ollivanders/src/net/pottercraft/Ollivanders2/StationarySpell/REPELLO_MUGGLETON.java]: ...

**Pull request merged**